controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -947,10 +948,11 @@ func (r *BareMetalHostReconciler) SetupWithManager(mgr ctrl.Manager) error {
 						return true
 					}
 
-					if oldHost.Status.ErrorCount != newHost.Status.ErrorCount {
-						//skip reconcile loop
+					//Discard updates that do not increase the resource Generation (such as on Status.LastUpdated), except for the finalizer
+					if e.MetaNew.GetGeneration() == e.MetaOld.GetGeneration() && reflect.DeepEqual(e.MetaNew.GetFinalizers(), e.MetaOld.GetFinalizers()) {
```

Comment:
We need to make sure we react to annotations changing as well. AIUI that's part of the metadata so it doesn't affect the generation.

Reply:
Thanks, will do